sdk/txbuild/create_escrow.go

```diff
@@ -1,47 +1,57 @@
 package txbuild
 
 import (
-	"github.com/stellar/go/amount"
 	"github.com/stellar/go/keypair"
 	"github.com/stellar/go/txnbuild"
 )
 
 type CreateEscrowParams struct {
-	Creator             *keypair.FromAddress
-	Escrow              *keypair.FromAddress
-	SequenceNumber      int64
-	InitialContribution int64
+	Creator        *keypair.FromAddress
+	Escrow         *keypair.FromAddress
+	SequenceNumber int64
+	Asset          txnbuild.Asset
+	AssetLimit     string
 }
 
 func CreateEscrow(p CreateEscrowParams) (*txnbuild.Transaction, error) {
+	ops := []txnbuild.Operation{
+		&txnbuild.BeginSponsoringFutureReserves{
+			SponsoredID: p.Escrow.Address(),
+		},
+		&txnbuild.CreateAccount{
+			Destination: p.Escrow.Address(),
+			// base reserves sponsored by p.Creator
+			Amount: "0",
+		},
+		&txnbuild.SetOptions{
+			SourceAccount:   p.Escrow.Address(),
+			MasterWeight:    txnbuild.NewThreshold(0),
+			LowThreshold:    txnbuild.NewThreshold(1),
+			MediumThreshold: txnbuild.NewThreshold(1),
+			HighThreshold:   txnbuild.NewThreshold(1),
+			Signer:          &txnbuild.Signer{Address: p.Creator.Address(), Weight: 1},
+		},
+	}
+	if !p.Asset.IsNative() {
```

Comment:
this is the main change of this PR. If the channel asset is non-native, we need to add a trustline to the escrow account for it. If it's native, no operation is needed

Reply:
this context too, where the main change is pointed out is super super helpful! 💯